File: solutions/python/allergies/1/allergies.py
class Allergies:
    ALLERGENS = [
        'eggs',
        'peanuts',
        'shellfish',
        'strawberries',
        'tomatoes',
        'chocolate',
        'pollen',
        'cats'
    ]

    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        return item in self.lst

    @property
    def lst(self):
        res = []
        for i, allergen in enumerate(self.ALLERGENS):
            if (self.score >> i) & 1:
                res.append(allergen)
        return res

File: solutions/ruby/allergies/1/allergies.rb
class Allergies
  ITEMS = %w[eggs peanuts shellfish strawberries tomatoes chocolate pollen cats]

  def initialize(score)
    @score = score
  end

  def allergic_to?(item)
    idx = ITEMS.index(item)
    @score & (1 << idx) != 0
  end

  def list
    ITEMS.select { |item| allergic_to?(item) }
  end
end
